### ISSUE TITLE
Duva Client feedback on

### ISSUE THREAD

**Issue by akahure** [onaio/duva#22]
Feedback from the client on https://github.com/onaio/duva/issues/18

It’s great the data columns can be grabbed with their “labels” instead of “names”. However, the same approach needs to be made available for the “data/submissions” as well.

Briefly saying, we quite need the same “export options” to be available for the connector. If not all of them, these are the most needed options:

- Column names: questions’ labels instead of names (DONE)
- Choice labels instead of names
- Split multiple answers into separate columns

Adding the other options (available in the export options) will be great but less priority.

It would be super useful if the above options are made as part of the configuration so users can enable/disable as needed (just the same as Export Options)

On the other hand, below are minor UI/UX issues for improvement:

- When adding the connector, a new page (for authorization) is open before choosing (add or use existing connection)
- The Edit button gives only the option for deletion (configs still can’t be edited)
- The Refresh icon doesn’t behave like a button (hand pointer on hover)
- Moving Token Value just below Token Name (next to each other)
- The “Use existing connection” menu gives one EMPTY entry
- Suggestion: Adding link to the corresponding Help article
- Suggestion: Adding a guidance Tool-Tip, with a screenshot image, to each field/datapoint of the configurations (site, token, site name, ..)

Please let me know your thoughts on the above feedback.

**Comment by akahure** [comment]
@DavisRayM lets chat about this on Monday and prepare feedback

**Comment by DavisRayM** [comment]
@akahure Adding the export configurations to the configuration API won't be too hard; Will require a few changes to support it but should take at most 2-3 days to do so(Pessimistic estimate). Enabling the export features globally will take a shorter time and will just be a small code change to ensure we pass those options to OnaData when requesting a CSV Export

The UI/UX changes seem to be more for the frontend not sure of the level of effort there; The API does provide a way to edit configurations.

**Comment by msschroeder** [comment]
@akahure These seem like feature request - not bugs? Therefore, we need to consider them as a part of the either the project or as a feature request, which would be decided in the quarterly roadmaps?

**Comment by akahure** [comment]
@ukanga based on discussion done with @denniswambua and @DavisRayM we outlined from the client requested changes we need to: 
Make updates to the Export API for Ona data to include the option to have choice labels instead of names. The tasks we highlighted are:

- We need to make UI updates on the Duva APP page to include the export config and edit connection functionality. To be done on the below page
![Screenshot - 2021-09-24T122641 719](https://user-images.githubusercontent.com/43808025/134651998-52818199-3222-41fb-a324-ae56dfed3692.png)

i.e. when a user clicks the export option 
- They should have the option to 'check' Choice labels and split multiple answers into separate columns as part of the export. This is proposed to be done on the DUVA settings page and have the export options page: https://prnt.sc/1t85vye  

Estimated LOE 
UI updates - 4 days 
Settings for export to be added to Duva page - 4 days 
QA -1 day 

On some of the UI Updates we need to: 
- Move Token Value below the token name 

![Screenshot - 2021-09-24T123956 112](https://user-images.githubusercontent.com/43808025/134653870-24904ec7-ea42-4c24-bc44-f567edbdf4f2.png)

- Add a link to the corresponding help article on the Duva settings page 
- Maybe have a guidance tool tip maybe per field?
